### ISSUE TITLE
Custom item models not showing

### ISSUE THREAD

**Issue by mrcoffee1026** [eccentricdevotion/TARDIS-Resource-Pack#1]
Not sure what the key off is for building the new version of the tardis which consists of one block and a trap door... but perhaps it should be more than just "a block".  I tried breaking a few to make some of them vanish and this crashed my grame.

This seems to adversely affect mushroom blocks, I'm not sure if anything else, but this is most noticeable in our spawn area.

A couple of examples:
https://imgur.com/a/eYgPvWo

Crash log if it helps:
https://pastebin.com/bHyqMAie

**Comment by mrcoffee1026** [comment]
Might matter:  grown mushrooms look fine.  The mushrooms in our spawn were built at some point though i'm fairly certain all the blocks in place on my server were copied with worldedit.

**Comment by eccentricdevotion** [comment]
There seems to be an issue with Mushroom blocks that have been placed already.
Blocks placed after the recent update should be OK.
See: https://github.com/eccentricdevotion/TARDIS/issues/125
and: https://www.youtube.com/watch?v=Ntjx-lARF5E

**Comment by mrcoffee1026** [comment]
So basically i'd have to rebuild these mushrooms to fix them... and breaking them crashes the game.  No chance a //replace with a specific block state would resolve this?

**Comment by eccentricdevotion** [comment]
Supposedly the [technique to use "unused" mushroom block states](https://www.curseforge.com/minecraft/texture-packs/bonus-mushroomblocks-add-160-custom-blocks) doesn't take into account blocks that don't occur naturally...
`//replace` might work, I don't have an existing scenario to test...

**Comment by eccentricdevotion** [comment]
Not sure why breaking them would cause a game crash, not seen anything like that in my testing. Is there anything in server / client logs that points to TARDIS?

**Comment by eccentricdevotion** [comment]
Can you make a WorldEdit schematic of the affected area and attach to this issue?

**Comment by mrcoffee1026** [comment]
I tried this to one of them it "worked", kinda:
//replace minecraft:brown_mushroom_block minecraft:brown_mushroom_block[up=true,north=true]
Turned it into the fancy texture i believe you use for siege though instead of the normal mushroom block.  I'll see if I can get you a schem of one before I do anything to it and tack it here next.

**Comment by eccentricdevotion** [comment]
`minecraft:brown_mushroom_block[up=true,north=true,south=true,west=true,east=true,down=true]`
I would change all the faces...

**Comment by mrcoffee1026** [comment]
Tried all faces, this did not make any change in the final appearance.  Here's a schem of one: 
[mushroomtree.zip](https://github.com/eccentricdevotion/TARDIS-Resource-Pack/files/3736306/mushroomtree.zip)

**Comment by mrcoffee1026** [comment]
It looking like the seige block isn't the worst thing that could happen but if there's a way to properly select the mushroom block i'd do that instead.

**Comment by eccentricdevotion** [comment]
`//replace minecraft:brown_mushroom_block minecraft:brown_mushroom_block[down=true,east=true,north=true,south=true,up=true,west=true]`
and
`//replace minecraft:mushroom_stem minecraft:mushroom_stem[down=true,east=true,north=true,south=true,up=true,west=true]`
works for me

**Comment by eccentricdevotion** [comment]
![mushroomtree](https://user-images.githubusercontent.com/1841924/66958224-37af4380-f0c4-11e9-94d7-e0075c38c5ee.jpg)

**Comment by mrcoffee1026** [comment]
when i do that they look this for me: https://imgur.com/a/mfiN4bV

**Comment by eccentricdevotion** [comment]
I'd say you have some of the old resource pack files still in place, you need to completely remove TARDIS-MCP

**Comment by mrcoffee1026** [comment]
yeah that's it.  Thanks much!

**Comment by mrcoffee1026** [comment]
Another problem though - if I completely remove the TARDIS-MCP... as the item textures don't appear to work in the new resource pack... keys, sonics, food no longer has textures.  I'm looking through the structure those items seem to "exist" they are just not being used.  Any idea why that might be?

**Comment by eccentricdevotion** [comment]
Because it's a new resource pack structure / features, the textures won't be applied to old items.
You should be able to use the sonic / key prefs menu to update those items, and any new food items and other items should have textures when crafted.
You could perhaps just use the TARDIS-MCP pack with only the Optifine CIT folder until all items are used up, but I haven't tested it.
It would also be possible to create a command that would update old items so that they use the new textures, ~~though it might be a bit of a pain to implement~~.

**Comment by mrcoffee1026** [comment]
i might try it with the cit folder since not having this also removes all of the icons in the menus within the tardis itself, and those don't get "recrafted".... i'm pretty sure the only thing I don't want from the original resource pack is just the mushroom block textures themselves or nearly nothing will look right.

**Comment by eccentricdevotion** [comment]
TARDIS GUI menus should all be working without CIT, as they get created when they are opened.

**Comment by mrcoffee1026** [comment]
This is what it's looking like with the new resource pack without any further modifications:
https://imgur.com/a/rHOKcsa

**Comment by eccentricdevotion** [comment]
All normal here: https://www.youtube.com/watch?v=T3gaOOBQl7E
Vanilla client (no Optifine installed), resource pack as it is on GitHub, no other resource packs except TARDIS Sounds.

**Comment by eccentricdevotion** [comment]
TARDIS build 1917 adds a `/tardis item [hand|inventory]` command to update old TARDIS items to be new resource pack compatible.

**Comment by mrcoffee1026** [comment]
whether the items are new or not does not seem to mater.  Old tardis keys and new tardis keys are the same item, so they don't have textures if i use the new resource pack (same with foods).  I'll try with the new build and see if that changes anything, but as far as I can tell, I'll have to combine the resource packs together into a merged one in order to have the textures displaying properly... won't know without several restarts at this point though.

**Comment by eccentricdevotion** [comment]
Reinstall the resource pack? Sounds like it needs it.
https://www.youtube.com/watch?v=EDOCUx847ik

**Comment by mrcoffee1026** [comment]
copying the CIT folder from the MCP resource pack made everything look right again.  Reinstalling isn't working.  I combine the texture packs: Sound Resource, Weeping Angels, and this new one and then install the result as the server's texture pack.  The textures for those items are just not in there anymore since I'm not including MCP in the bundle.  I tried just using the new one on its own locally and got the same results, so it's not a problem with the bundle.  Also attempted the new resource pack locally along with command: "The item already has custom model data!".  The tardis keys look like remotes while in your hand but still like nuggets in inventory.  Using the command on custard makes it look properly in my hand but not in my inventory.  The command says it updated 15 items in my inventory... but doesn't change the way any of them display.  No commands work on the tardis menus since they aren't actually in my hand or my inventory, so they still look like vanilla redstone and wooden bowls.

**Comment by eccentricdevotion** [comment]
Can you upload a zip file of the pack you are using somewhere so I can check it out? Think that's the only way I can troubleshoot it.
The whole point of the new pack is to be able to see TARDIS menus without the use of Optifine...

**Comment by eccentricdevotion** [comment]
The new resource pack already has TARDIS Weeping Angel textures in it, no need to combine it with the standalone TWA resource pack.

**Comment by mrcoffee1026** [comment]
oh huh, that's interesting.  Does having optifine ruin it then?  As there are things optifine does outside of rending the texture pack which I'd want to keep using it for.  If having optifine prevents it from displaying properly though, that's kind of a deal-breaker since I just spent the last two years talking most players into installing and using Optifine.  My combined zip is here although the only difference right now is the included CTI folder from the original MCP resource pack... the documentation on all the github pages for these should really be updated so I know what to get these days, they all still say to get that, the weeping angels, the sound resource pack and also the bromley vector pack (which i don't since that changes way, way too much).
Too large for upload here, the one i'm using is http://communitybrew.net/downloads/TARDIS.zip
Note: this one is working as I would have expected it to have.  The new one as provided did not.
Has the CTI folder from MCP and the sound/weeping angels combined (I just didn't copy anything that was a duplicate for the weeping angels stuff, I think it skipped most if not all of it).

**Comment by eccentricdevotion** [comment]
The new pack still uses some Optifine features so is totally compatible (my Minecraft client has it installed by default, but it will work with or without). But for those that don't want to use Optifine, the new pack provides textures for items and GUIs by utilising custom model data specified in JSON files.
I'll download your pack now and see if I can find what might be wrong.

**Comment by eccentricdevotion** [comment]
Your pack is working for me __with and without the CIT folder__. Even with the CIT folder in the pack, I still see the custom model for things like the Jammy Dodgers instead of the plain CIT ones (both in hand and in the inventory, and the pack loaded locally or as a server pack, Optifine enabled and disabled).
So doesn't seem to be an issue with the pack itself...

**Comment by mrcoffee1026** [comment]
In my pack I added the CIT folder in is why.  I started out using only the new resource pack (+ sounds and I guess + weeping angels which was redundant) and the result was as I described: no menus styled, no foods or items textured.  When I used the commands... it would style the tardis key incorrectly as a remote control while holding it.. and not at all within my inventory.  The command to style things within my inventory, while it echoed confirmation, did nothing other than say it did something.  The textures did not update and the items still looked as they do without the resource pack.  I would remove the CIT folder as you suggest here but doing so basically unstyles nearly everything used in the plugin.  I think handles still shows up properly, and that's about it.

**Comment by eccentricdevotion** [comment]
Yup, I installed your pack but _removed the added CIT folder_ and I still see TARDIS items and menus with the correct models and textures using the new system. Which is why I think it's not the pack that is at fault. Are you using any other mods? Are you using the default Minecraft launcher?

**Comment by eccentricdevotion** [comment]
If you turn on the output log in the Minecraft launcher are there any resource pack related errors in the log (apart from 4 `Unable to load model: 'minecraft:block/chemistry/xxx_torch'`)?

**Comment by eccentricdevotion** [comment]
Are you using any other (non-TARDIS) resource packs?

**Comment by mrcoffee1026** [comment]
I am using the default mc launcher, forge, optifine, betterpvp (mod) and also the faithful (r4.1) resource pack as well as this one.  Well no errors in the log with the pack as i have it now, but i'll try removing the folder and see what happens maybe later today (i have an eye appointment and will be gone most of the day so won't get a chance to try it till later).  I don't think it's the faithful pack doing it since it since all other changed items like blocks or items get replaced by whatever other pack is on the server, but let me know either way.

**Comment by eccentricdevotion** [comment]
Still can't reproduce your item problems - I've installed Forge / BetterPVP / Optifine / Faithful and see the correct TARDIS items. Have tried _both_ loading locally and a server resource pack.

**Comment by JasonPra** [comment]
What I don't get is why some of the gui textures work wile most don't. I am also having problems with all the custom item modles...
![Screenshot 2019-10-20 04 59 31](https://user-images.githubusercontent.com/13491965/67157808-01a5d580-f2f7-11e9-9205-741e4ec24e6d.png)

**Comment by mrcoffee1026** [comment]
alright well if it's some kind of internal optifine caching it appears to be completely unclearable, so will happen to other players if I don't include the optifine folder.  So long as I know to do that, it's fine and I guess some limitation of the framework that cannot be fixed but at least there's a work around.  I'd like the final pack to be smaller in file size than it is but it's mostly tolerable.

**Comment by JasonPra** [comment]
So you planing to do anything with the 3d tardis shells?

**Comment by eccentricdevotion** [comment]
What do you mean? They’re already incorporated into the TARDIS Chameleon Circuit and can be used for the TARDIS exterior...
https://youtu.be/dZORqsoI90U

**Comment by JasonPra** [comment]
What am i missing i have the resource pack and optifine but most of the textures don't work. They seem to be working just fine for you.

**Comment by JasonPra** [comment]
Nevermind had a older texture pack

**Comment by eccentricdevotion** [comment]
@mrcoffee1026 
Fixed a couple of minor JSON errors, can you see if it makes any difference on your setup.
https://github.com/eccentricdevotion/TARDIS-Resource-Pack/releases/tag/1.0.0

**Comment by mrcoffee1026** [comment]
Same exact thing.  All I did was download the latest resource pack file from github and combined ONLY the sound resource pack assets into it.  Same problem with the new pack: https://imgur.com/a/mB357RW  Will keep having to use the MCP folder to fix and repair all of the missing elements for optifine.


Even with this folder, since you pointed out the redundancy and I only had to include that folder and the sound package, the final package is still smaller by about 5MB which is helpful.  I do appreciate that you've updated the information on the github pages about these resource packs to help with the confusion as well.  Ah nevermind it's going to be larger since the sound resource pack doesn't include strax and a few other things, i"ll have to grab those sounds from weeping angels and recombine and it'll be the same thing all over again.  Oh well, it would have been easier if the sound pack was actually complete, that's unfortunate.

Final modified one I'm using now is still quite a ways smaller:
(http://comminutybrew.net/downloads/TARDIS.zip)
This is the new resource pack + the sound reource pack + the sounds from the weeping angels pack + the cit folder from the MCP's optifine folder's pack.  The new resource pack already included a cit folder for optifine so I did not copy anything into there that was already present there, only the items that were not there already.  This will work as a workaround for now, as far as why the new textures aren't appearing properly... if there's some sort of optifine cache keeping the wrong textures in place or something, there doesn't seem to be any way to clear them.  They automatically display wrongly whenever I attempt any version of the new resource pack that doesn't have the cit stuff from the MCP pack.

**Comment by mrcoffee1026** [comment]
Since Optifine is not an option in 1.15 and won't be for awhile it might be a good time to revisit some of the more lingering issues here.  The internal menus of the tardis - I have found - without Optifine installed while running 1.15 using the 1.15 update: finally display properly.  The tardis "key" item is a rather uglier shade of almost orange without the old optifine models, but otherwise things look pretty much proper.  The tardis weeping angel mobs do NOT, however.  AT all.  Wolfs are wolfs again, there are no K-9s and no ability to fix any of the mobs without Optifine being installed.  The primary Tardis-Resource-Pack is still lacking all of the sounds from the TARDIS-Weeping-Angels-Resource Pack.  Custom wool textures are now just wool as those changes were only ever saved as Optifine models and never added to the new resource pack.  Since the later update to the texture pack I never did discover how to re-access the Ood models.  If there's some magic to converting Vindicators to Ood now, I think we'd all like to know about it.

https://imgur.com/a/ZqAgRHo

Pictures feature "orange wool" which used to have bolt-like "round things' on them from the original resource pack, but not without optifine.  (Included pic of outside of TARDIS as evidence I am indeed using the latest resource pack)  Also note in the Deluxe tardis I have swapped the uglified 1.14 half stone slabs with smooth stone slabs which I highly advise for the schematic (it was what the block was originally, the update to 1.14 was very negative for anyone who used the old block for anything).

**Comment by eccentricdevotion** [comment]
TARDIS keys are not meant to be orange, but brass or silver coloured. There is a gold coloured key texture which could be used instead of the ugly 3D modelled one which I don't really like.

For TARDIS Weeping Angel monsters, we are limited by the current capabilities of Minecraft itself, one of three things would need to change:

1. Ability to put an item into the chestplate / leggings / boots armour slots and have it render in-game, or
2. Ability to change armour models, or
3. Ability to add custom  mobs to Java Edition (currently possible in Bedrock)

![cyber_armour](https://user-images.githubusercontent.com/1841924/71550331-c6303180-2a32-11ea-9f28-5fed82ac88fa.png)

TARDIS Weeping Angels still has its own (updated) resource pack with Ood illagers, feel free to combine resource packs as you wish.

Custom TARDIS interior blocks could be changed, I think there are still a few unused mushroom block states that could have the current custom textures applied.

Will update the deluxe, eleventh and twelfth consoles with smooth stone slabs.

**Comment by mrcoffee1026** [comment]
That resource pack (https://github.com/eccentricdevotion/TARDISWeepingAngels-Resource-Pack) last updated 4 months ago and that's what we were using before upgrading to 1.14.  It was when in 1.14 I discovered that we couldn't see the Ood anymore... that model used to be applied to ALL vindicators... and ceased doing so with that version.  If there's some way of getting them to look like that again, it isn't documented anywhere, but I would assume they must need a particular name applied to them before that model will be applied.  That update also ceased modeling all skeletons as weeping angels and all endermen as the silence... and properly only modified the proper mobs to look that way, suggesting that there should have been a similar method employed to only utilize K-9 when desired instead of eliminating the entire species of flesh and blood wolves from the entire server by using this resource pack.

If the custom interior blocks were changed, the schematics weren't updated to use the  new blocks you've decided to use instead.  However... the orange wool block had its own special texture and was the default wall color in virtually every example given, I doubt it was changed to have "no texture", I believe a number of blocks were simply overlooked in the transition to the newer TARDIS-Resource-Pack.  I didn't go through and test the various mushroom blocks... there was also some odd configuration of furnaces that would result in a number of blocks that appeared like they had red buttons on them... but I'm not sure what the necessary configuration actually was, I just stumbled across a wall with this appearance in a random build and couldn't duplicate it... that very appearance may have only worked in optifine I'm not certain.  I haven't stumbled across any blocks like it while using the actual TARDIS plugin, I'm just confident it was the source of the images, but I don't know if they're actually needed or used for anything, but they were around in the 1.14 version of the pack.

**Comment by eccentricdevotion** [comment]
Textures for illagers (red-eyed Ood) and endermen (Silence) are still there, unchanged for at least 2 years (4 years in the case of endermen). Regular Ood as villagers are kind of in the pack, but you would have to change the filename to get them to work, as they were never really implemented anyway...

![vindicator](https://user-images.githubusercontent.com/1841924/71551367-a442a900-2a4a-11ea-8043-10fd5ac98258.jpg)

https://github.com/eccentricdevotion/TARDISWeepingAngels-Resource-Pack/tree/master/assets/minecraft/textures/entity/illager
https://github.com/eccentricdevotion/TARDISWeepingAngels-Resource-Pack/tree/master/assets/minecraft/textures/entity/enderman

Weeping angels no longer need to override the skeleton texture as they now have a custom block model for their head, and Optifine takes care of the rest with custom armour textures.

K-9 uses an Optifine custom entity model to override the wolf model so won't work currently.

I haven't changed the schematics, they have always used orange wool. It relies on Optifine to change the texture using Connected Textures based on the biome and as such have not been "overlooked". Using mushroom blocks for them is on the TODO list...

The resource pack doesn't override any furnace block models, only item models displayed in GUIs. Artron furnaces use Optifine & CTM. In any case, furnaces don't have any red buttons associated with them...

**Comment by mrcoffee1026** [comment]
Ok sorry didn't realize those blocks were on the todo list still, when the "TARDIS-Resource-Pack" became first available and I asked about it you'd said it was designed to work without Optifine and to replace things styled by the old packs, I took that to mean it was complete by itself.

Well I didn't have to change the file names to see the Ood in the earlier version... and then they vanished.  It's just as well then since I would rather they just not be present than have all Vindicators suddenly be Ood, as they do not at all behave as Ood.  I just wish some of the mobs like k-9 could be a bit more selective when being use... again, I realize it's an Optifine thing but since it was finally able to finally differentiate between normal Skeletons and Weeping angels... and differentiate between Endermen and The Silence... it should also be able to differentiate between normal dogs and K-9s.

And I get that Optifine is required for the entirety of everything Weeping Angels related, that's just... too bad since it's probably going to be awhile... no progress in a week and more holiday on the way.

**Comment by eccentricdevotion** [comment]
Ood - https://dev.bukkit.org/projects/tardisweepingangels?comment=212
Neither the resource pack or Optifine differentiates between skeletons and weeping angels / endermen and silence - weeping angels are still just skeletons underneath the armour and always have been. Even with Optifine installed all endermen will be silence as the texture file just overrides the default enderman skin - there is no plugin / mod magic going on there. The only difference between endermen and silence is that the silence have an invisible guardian passenger (which is plugin related). I'm not sure why you don't see the textures if they are in your resource pack...
Anyway we'll get there one day, it'll just take a lot longer than waiting for Optifine to update...

**Comment by mrcoffee1026** [comment]
that's very strange since for the last few months we've been enjoying normal endermen/skeletons as well as fully skinned Silence and Weeping Angels... I thought it was an update to optifine made one of the later weeping angels updates was taking advantage of.

**Comment by eccentricdevotion** [comment]
TARDIS v4.3 build 1943

[![TARDIS interior walls](https://i.ibb.co/B4qKs9d/https-i-ytimg-com-vi-QBhpvrisi-LI-maxresdefault.jpg)](https://youtu.be/QBhpvrisiLI "TARDIS interior walls")

1d5be67

**Comment by mrcoffee1026** [comment]
Not working like that at all, currently.  Orange wool looks like orange wool.

**Comment by eccentricdevotion** [comment]
You would need to update the TARDIS plugin on the server, and update the resource pack, changes have been pushed to this repository 1d5be67
Existing blocks will still be Orange Wool (hexagon) / White Stained Glass (roundel) / White Terracotta (offset roundel) - you would need to run the desktop theme to update existing blocks so that they are changed to mushroom stems with a custom block model and textures. Blocks in existing rooms won't be updated as their is no TARDIS feature that changes the walls.

**Comment by eccentricdevotion** [comment]
Added TWA sounds to the TARDIS-SoundResourcePack
https://github.com/eccentricdevotion/TARDIS-SoundResourcePack/commit/1777d78131d882a026fe0df6adc24e976a710112

**Comment by mrcoffee1026** [comment]
Ok yeah a reset wasn't good enough either, it still maintained the orange wool walls.  Guess this means the junk tardis has to be destroyed and rebuilt to look correct also.

**Comment by eccentricdevotion** [comment]
Actually I completely forgot the Junk TARDIS, will do that now

**Comment by mrcoffee1026** [comment]
oh ok and then let me know how to fix it if there's a different method....

**Comment by mrcoffee1026** [comment]
we really appreciate all the little fixes, by the way

**Comment by eccentricdevotion** [comment]
TARDIS build 1944
Non-Optifine wall textures for Junk TARDIS...

![junk_tardis_no_optifine](https://user-images.githubusercontent.com/1841924/71572849-e997d100-2b45-11ea-96a8-cada8ea132a6.jpg)

... and Junk Mode

![junk_mode_no_optifine](https://user-images.githubusercontent.com/1841924/71572848-e997d100-2b45-11ea-8877-576cf6606903.jpg)

Better TARDISWeepingAngel monsters when Optifine is not installed - TARDISWeepingAngels build 123 - Monsters should update themselves to the new heads and arms when the chunk they are in loads.

![twa_mosters_no_optifine](https://user-images.githubusercontent.com/1841924/71572850-ea306780-2b45-11ea-895e-4f1c0614841e.jpg)

**Comment by mrcoffee1026** [comment]
What do I have to do to fix the junk tardis?  I did a delete and create of the thing but it came back as the same vanilla orange wool texture and I'm not sure what else to do to affect its change?

**Comment by eccentricdevotion** [comment]
Update TARDIS plugin (build 1944), update resource pack (last update 17 hours ago, though [this one](https://github.com/eccentricdevotion/TARDIS-Resource-Pack/releases/tag/1.0.1) should work OK)

**Comment by mrcoffee1026** [comment]
an somehow i neglected to update the tardis build to 1944 on the server, certainly downloaded it, just didn't also upload it, that should fix it then, thanks.

**Comment by mrcoffee1026** [comment]
yes that does fix it, thank you!

I think the only issues that MIGHT remain would be from any sort of interference from optifine when it is installed, but there've been so many changes since I first opened the ticket that if that is still an issue, it is not likely to be the same issue that I started out with.  As of now, I have some weird thing going on where I cannot see beacons, and I suspect that previous Optifine installs might leave behind settings or other traces that might cause issues like that even when it is not "installed" afterwards as the beam of light is not altered by resource packs that I am aware of.  In any case this ticket has been open long enough on this topic and since it isn't bothering us currently is good to close.
